vpr/src/server/zlibutils.cpp

```diff
@@ -0,0 +1,83 @@
+#ifndef NO_SERVER
+
+#include "zlibutils.h"
+
+#include <cstring> // Include cstring for memset
+#include <zlib.h>
+
+std::optional<std::string> try_compress(const std::string& decompressed) {
+    z_stream zs;
+    memset(&zs, 0, sizeof(zs));
+
+    if (deflateInit(&zs, Z_BEST_COMPRESSION) != Z_OK) {
+        return std::nullopt;
+    }
+
+    zs.next_in = reinterpret_cast<Bytef*>(const_cast<char*>(decompressed.data()));
+    zs.avail_in = decompressed.size();
+
+    int ret_code;
+    char* result_buffer = new char[32768];
```

Comment:
Magic number. Consider making it a variable or a comment saying the buffer is 32 KB in size.

Reply:
fixed https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/2526/commits/f35de4ee90058597f73ddbc3d40d68abe9593d70